manager/controlapi/service.go

```diff
@@ -26,55 +26,88 @@ func validateResources(r *api.Resources) error {
 		return nil
 	}
 
+	var me MultiError
+
 	if r.NanoCPUs != 0 && r.NanoCPUs < 1e6 {
-		return grpc.Errorf(codes.InvalidArgument, "invalid cpu value %g: Must be at least %g", float64(r.NanoCPUs)/1e9, 1e6/1e9)
+		err := grpc.Errorf(codes.InvalidArgument, "invalid cpu value %g: must be at least %g", float64(r.NanoCPUs)/1e9, 1e6/1e9)
+		me = append(me, err)
 	}
 
 	if r.MemoryBytes != 0 && r.MemoryBytes < 4*1024*1024 {
-		return grpc.Errorf(codes.InvalidArgument, "invalid memory value %d: Must be at least 4MiB", r.MemoryBytes)
+		err := grpc.Errorf(codes.InvalidArgument, "invalid memory value %d: must be at least 4MiB", r.MemoryBytes)
+		me = append(me, err)
 	}
-	return nil
+
+	if len(me) == 0 {
+		return nil
+	}
+
+	return me
 }
 
 func validateResourceRequirements(r *api.ResourceRequirements) error {
 	if r == nil {
 		return nil
 	}
+
+	var me MultiError
+
 	if err := validateResources(r.Limits); err != nil {
-		return err
+		me = append(me, err)
 	}
+
 	if err := validateResources(r.Reservations); err != nil {
-		return err
+		me = append(me, err)
 	}
-	return nil
+
+	if len(me) == 0 {
+		return nil
+	}
+
+	return me
 }
 
 func validateRestartPolicy(rp *api.RestartPolicy) error {
 	if rp == nil {
 		return nil
 	}
 
+	var me MultiError
+
 	if rp.Delay != nil {
 		delay, err := ptypes.Duration(rp.Delay)
 		if err != nil {
-			return err
-		}
-		if delay < 0 {
-			return grpc.Errorf(codes.InvalidArgument, "TaskSpec: restart-delay cannot be negative")
+			me = append(me, grpc.Errorf(codes.InvalidArgument, "TaskSpec: restart-delay parse error: %v", err))
+		} else {
+			if delay < 0 {
+				err = grpc.Errorf(codes.InvalidArgument, "TaskSpec: restart-delay cannot be negative")
+				me = append(me, err)
+			}
 		}
 	}
 
 	if rp.Window != nil {
 		win, err := ptypes.Duration(rp.Window)
 		if err != nil {
-			return err
-		}
-		if win < 0 {
-			return grpc.Errorf(codes.InvalidArgument, "TaskSpec: restart-window cannot be negative")
+			me = append(me, grpc.Errorf(codes.InvalidArgument, "TaskSpec: restart-window parse error: %v", err))
+		} else {
+			if win < 0 {
+				err = grpc.Errorf(codes.InvalidArgument, "TaskSpec: restart-window cannot be negative")
+				me = append(me, err)
+			}
 		}
 	}
 
-	return nil
+	if rp.Condition != api.RestartOnNone && rp.Condition != api.RestartOnFailure && rp.Condition != api.RestartOnAny {
+		err := grpc.Errorf(codes.InvalidArgument, "TaskSpec: restart-condition should be in range 0-2")
```

Comment:
I'd prefer not to mention protobuf field numbers in error messages. They won't be meaningful to the user.